src/dataframe/dataframe.jl

```diff
@@ -625,6 +625,45 @@ Base.getindex(df::DataFrame, row_ind::Colon, col_inds::MultiColumnIndex) =
 Base.getindex(df::DataFrame, row_ind::typeof(!), col_inds::MultiColumnIndex) =
     select(df, index(df)[col_inds], copycols=false)
 
+# Gets a subset of a DataFrame by `row_inds` and disallows missing values in columns specified by `col_disallowmissing_inds` 
+# Called by dropmissing()
+function _getindex_disallowmissing(df::DataFrame, row_inds::AbstractVector{T},
```

Comment:
what if `df` is not a `DataFrame`?

Reply:
Excellent point! I've been too focused on the main case.

Would it make sense to just rely on each type's own methods and do a general fallback?